packages/core/mesh/network-manager/src/transport/libdatachannel-transport.ts

```diff
@@ -0,0 +1,180 @@
+  private readonly _peer: Promise<PeerConnection>;
+  private _channel!: RTCDataChannel;
+  private _stream!: Duplex;
+
+  private readonly _readyForCandidates = new Trigger();
+  private _writeCallback: (() => void) | null = null;
+
+  constructor(private readonly params: LibDataChannelTransportParams) {
+    this._peer = (async () => {
+      const { PeerConnection } = await import('./datachannel/rtc-peer-connection');
+      const peer = new PeerConnection(params.webrtcConfig);
+
+      peer.onicecandidateerror = (event) => {
+        log.error('onicecandidateerror', { event });
+      };
+
+      peer.onconnectionstatechange = (event) => {
+        log.info('onconnectionstatechange', { event, state: peer.connectionState });
+      };
+
+      peer.onicecandidate = async (event) => {
+        if (event.candidate) {
+          await params.sendSignal({
+            payload: {
+              data: {
+                type: 'candidate',
+                candidate: {
+                  candidate: event.candidate.candidate,
+                  sdpMLineIndex: event.candidate.sdpMLineIndex,
+                  sdpMid: event.candidate.sdpMid,
+                },
+              },
+            },
+          });
+        }
+      };
+      if (params.initiator) {
+        peer
+          .createOffer()
+          .then(async (offer) => {
+            await peer.setLocalDescription(offer);
+            await params.sendSignal({ payload: { data: { type: offer.type, sdp: offer.sdp } } });
+          })
+          .catch((err) => {
+            log.catch(err);
+          });
+
+        this._channel = peer.createDataChannel('dxos.mesh.transport');
+
+        this._channel.onopen = () => this._handleChannel(this._channel);
+      } else {
+        peer.ondatachannel = (event) => {
+          this._channel = event.channel;
+          this._channel.onopen = () => this._handleChannel(this._channel);
+        };
+      }
+
+      return peer;
+    })();
+  }
+
+  private _handleChannel(dataChannel: RTCDataChannel) {
+    this._channel.onerror = async (err) => {
+      log.error('channel error', { err });
+      await this._disconnectStreams();
+    };
+    this._channel.onclose = async (err) => {
+      log.error('channel error', { err });
+      await this._disconnectStreams();
+    };
+
+    const duplex = new Duplex({
+      read: () => {},
+      write: (chunk, encoding, callback) => {
+        // todo wait to open
+
+        dataChannel.send(chunk);
+
+        if (this._channel.bufferedAmount > MAX_BUFFERED_AMOUNT) {
+          this._writeCallback = callback;
+        } else {
+          callback();
+        }
+      },
+    });
+
+    dataChannel.onmessage = (event) => {
+      let data = event.data;
+      if (data instanceof ArrayBuffer) {
+        data = Buffer.from(data);
+      }
+      duplex.push(data);
+    };
+
+    // TODO(mykola): Check if this is working correctly.
+    dataChannel.onbufferedamountlow = () => {
```

Comment:
check if backpressure is actually working using the transport bench. add console logs here and where we start applyinh backpressure up top and verify that they are firing